trl/trainer/sft_trainer.py

```diff
@@ -293,7 +293,7 @@ def __init__(
         if args.padding_free:
             if data_collator is not None:
                 raise ValueError("Passing a custom data collator is not supported when using padding-free.")
-            if args.packing:
+            if args.packing and args.packing_strategy != "ffd":
```

Comment:
```suggestion
            if args.packing:
```

actually padding_free is a different method (which is less relevant once once we have ffd with flash-attn).